src/typescript/Scala.tmLanguage.ts

```diff
@@ -692,6 +705,20 @@ export const scalaTmLanguage: TmLanguage = {
             }
           ],
           name: 'meta.package.scala'
+        },
+        {
+          match: `\\b(given)\\s+(as)\\s`,
+          captures: {
+            '1': { name: 'keyword.declaration.scala' },
+            '2': { name: 'keyword.declaration.scala' }
+          }
+        },
+        {
+          match: `\\b(given)\\s+(${backQuotedId}|${plainid})?`,
+          captures: {
+            '1': { name: 'keyword.declaration.scala' },
+            '2': { name: 'entity.name.declaration' }
+          }
```

Comment:
This fails for the following unit test:

```scala
    given // this should be a comment
//  ^^^^^ keyword.declaration.scala
//        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ punctuation.definition.comment.scala
        as Foo
//      ^^ keyword.declaration.scala
```
The `//` is highlighted as an `entity.name.declaration` by the above rule. This is because `plainid` contains `(${opchar})+`, and `opchar` contains `/`. 

The solution may be to start a state on `(given)(\s+)`, that ends with `(?:(as)|(${backquotedId}|${plainid})?)`, and includes `'#comments'`. I don't really know how the optional capture group will work as an end marker though!

Reply:
I fixed the comment issue and joined the two rules into one. The optional captures just ignore the group if that part is not matched.